rust/parser/mod.rs

```diff
@@ -114,7 +114,7 @@ fn get_isa_constraint(
     match visit_type(ctx)? {
         Type::Unscoped(label) => Ok(IsaConstraint::from(label)),
         Type::Variable(var) => Ok(IsaConstraint::from(var)),
-        _ => Err(ILLEGAL_STATE.format(&[line!().to_string().as_str()])),
+        t => panic!("Unexpected type: {:?}", t),
```

Comment:
More descriptive panic message, I'd like to tighten `Type` further so that this isn't necessary altogether.